app/views/shared/address_selection/_no_address.html.erb

```diff
@@ -0,0 +1,27 @@
+<%= page_template(
+      page_title: t(".heading"),
+      template: :basic,
+      form:,
+    ) do %>
+
+      <h1 class="govuk-heading-xl"><%= page_title %></h1>
+
+      <p class="govuk-body">
+        <% if @form.model.building_number_name.present? %>
+            <%= t(".addresses_found_long_html",
+                  number: @address_collection.count,
+                  postcode: @form.postcode.insert(-4, " "),
+                  building_name_number: @form.model.building_number_name) %>
+          <% else %>
+            <%= t(".addresses_found_html",
+                  number: @address_collection.count,
+                  postcode: @form.postcode.insert(-4, " ")) %>
```

Comment:
Could we move `postcode.insert(-4, " ")` to the model or can we use the `pretty_postcode` method?